File: day1/chal2.py
f = open('chal1_input.txt')
floor = 0
i = 1
for line in f:
    for c in line:
        if c == "(":
            floor += 1
        elif c == ")":
            floor -= 1
        if floor < 0:
            print i
            break
        i += 1

File: unmigrated/2015/rust/day1/chal2.rs
use std::io::{BufReader,BufRead};
  use std::fs::File;
  use std::path::Path;

  fn main() {
      let path = Path::new("input.txt");
      let file = BufReader::new(File::open(&path).unwrap());
      for line in file.lines() {
        let mut count = 0;
        let unwraped = line.unwrap();
        for (i, c) in unwraped.chars().enumerate(){
          //print!("{} {}\n", c, i);
          if c == '(' {
            count += 1;
          }else if c == ')'{
            count -= 1;
          }
          if count < 0 {
            print!("entered basement at: {}\n", i+1);
            break;
          }
        }
      }
  }
